go/geth_adapter/adapter_test.go

```diff
@@ -466,3 +471,212 @@ func TestRunContextAdapter_bigIntToWord(t *testing.T) {
+		t.Errorf("Conversion returned wrong value, expected %v, got %v", tosca.R07_Istanbul, revision)
+	}
+}
+
+func TestRunContextAdapter_ConvertRevision(t *testing.T) {
+	cancunTime := uint64(1000)
+	shanghaiTime := uint64(900)
+	parisBlock := big.NewInt(100)
+	londonBlock := big.NewInt(90)
+	berlinBlock := big.NewInt(80)
+	istanbulBlock := big.NewInt(70)
+
+	tests := map[string]struct {
+		mergeBlock *big.Int
+		block      *big.Int
+		time       uint64
+		want       tosca.Revision
+	}{
+		"Istanbul": {
+			block: istanbulBlock,
+			time:  uint64(0),
+			want:  tosca.R07_Istanbul,
+		},
+		"Berlin": {
+			block: berlinBlock,
+			time:  uint64(0),
+			want:  tosca.R09_Berlin,
+		},
+		"London": {
+			block: londonBlock,
+			time:  uint64(0),
+			want:  tosca.R10_London,
+		},
+		"Paris": {
+			block: parisBlock,
+			time:  uint64(0),
+			want:  tosca.R11_Paris,
+		},
+		"Shanghai": {
+			block: parisBlock,
+			time:  shanghaiTime,
+			want:  tosca.R12_Shanghai,
+		},
+		"Cancun": {
+			block: parisBlock,
+			time:  cancunTime,
+			want:  tosca.R13_Cancun,
+		},
+	}
+
+	chainConfig := &params.ChainConfig{
+		ChainID:            big.NewInt(42),
+		IstanbulBlock:      istanbulBlock,
+		LondonBlock:        londonBlock,
+		BerlinBlock:        berlinBlock,
+		MergeNetsplitBlock: parisBlock,
+		ShanghaiTime:       &shanghaiTime,
+		CancunTime:         &cancunTime,
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			got := convertRevision(chainConfig, test.block, test.time)
+			if got != test.want {
+				t.Errorf("Conversion returned wrong value, expected %v, got %v", test.want, got)
+			}
+		})
+	}
+}
+
+func TestRunContextAdapter_gethToVMErrors(t *testing.T) {
+	gas := tosca.Gas(42)
+	otherError := fmt.Errorf("other error")
+	tests := map[string]struct {
+		input      error
+		wantResult tosca.CallResult
+		wantError  error
+	}{
+		"nil": {
+			input: nil,
+		},
+		"insufficientBalance": {
+			input:      geth.ErrInsufficientBalance,
+			wantResult: tosca.CallResult{GasLeft: gas},
+			wantError:  nil,
+		},
+		"OutOfGas": {
+			input:      geth.ErrOutOfGas,
+			wantResult: tosca.CallResult{},
+			wantError:  nil,
+		},
+		"stackUnderflow": {
+			input:      &geth.ErrStackUnderflow{},
+			wantResult: tosca.CallResult{},
+			wantError:  nil,
+		},
+		"stackOverflow": {
+			input:      &geth.ErrStackOverflow{},
+			wantResult: tosca.CallResult{},
+			wantError:  nil,
+		},
+		"invalidOpCode": {
+			input:      &geth.ErrInvalidOpCode{},
+			wantResult: tosca.CallResult{},
+			wantError:  nil,
+		},
+		"other": {
+			input:      otherError,
+			wantResult: tosca.CallResult{},
+			wantError:  otherError,
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			gotResult, gotErr := gethToVMErrors(test.input, gas)
+			if !errors.Is(gotErr, test.wantError) {
+				t.Errorf("Unexpected error: expected %v, got %v", test.wantError, gotErr)
+			}
+			if gotResult.Success != test.wantResult.Success {
```

Comment:
nit: could probably be tested using [`reflect.DeepEqual`](https://pkg.go.dev/reflect#DeepEqual). This would have the additional benefit that if the result struct gets altered in the future for whatever reason, the equality check would cover this.